src/Composer/Util/RemoteFilesystem.php

```diff
@@ -436,6 +449,9 @@ protected function getOptionsForUrl($originUrl, $additionalOptions)
             $options['http']['header'][] = $header;
         }
 
+        if($this->config && $this->config->get('gitlab-domains') && $originUrl == $this->config->get('gitlab-domains')[0]) {
+            $options['retry-auth-failure'] = false;
```

Comment:
It follows the [GitHub driver](https://github.com/composer/composer/blob/a066171dcc023ad6429881a7692e46f10e080f99/src/Composer/Util/GitHub.php#L106-L108).